pandas/core/frame.py

```diff
@@ -4037,6 +4049,8 @@ def set_index(self, keys, drop=True, append=False, inplace=False,
     def reset_index(self, level=None, drop=False, inplace=False, col_level=0,
                     col_fill=''):
         """
+        An existing index is modified.
```

Comment:
I think we can also do better here for the summary method, as just from this line "An existing index is modified.", a user won't get much clue of what the method is doing.

But of course, given that the method is doing different things, it might be difficult to have something that is still generally true but less vague as the above ... 

Trying to think of better descriptions:

`Remove index (level)` (but that is also a bit short / cryptic) 
`Reset index to default index or remove index level`

(will think further on it)